### DISCUSSION TITLE
Clarification on incremental clustering: What if the input file has stay_lat, stay_long and stay_dur columns?

### DISCUSSION THREAD

**Discussion by Anurag19101996** [uw-ssec/MAWpy#44] [General]
For the existing code system (older MAW version), the algorithm processes GPS coordinates (Stay_lat, Stay_long) if the stay_duration exceeds the specified duration constraint (given user provides a duration constraint), and this happens if incremental clustering is not the initial step in the workflow.

These GPS coordinates, Stay_lat and Stay_long, are then utilized for subsequent clustering operations.

My doubt is, since the Stay_lat and Stay_long are not a part of the new input file system and are created within the new code, how will the new code work if the user intends to provide the Stay_dur, Stay_lat and Stay_long columns in input? Could you share some insights on this?

@anujsinha3 @carlosgjs @qzchen-uw @gracejia513

**Comment by anujsinha3** [comment]
The new code behaves exactly how you mentioned:

If the user provides a duration constraint for the cases where incremental_clustering is not the first step, ONLY then does the code expect the 'stay_lat' and 'stay_long' to be present as columns.

If you check the code below, you can understand the behavior:

https://github.com/uw-ssec/MAWpy/blob/main/src/mawpy/steps/incremental_clustering.py#L288
